src/main/java/Duke.java

```diff
@@ -1,10 +1,167 @@
+            }
+        }
+    }
+
+    public static void handleCommand(String input) {
+
+    }
+
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("data/duke.txt").run();
+    }
+
+    // TODO: Refactor this part to be integrated with CLI UI
+    public String getResponse(String input) {
+        String toReturn;
+        try {
+            if (Parser.commandEquals("bye", input)) {
+                Ui.sayGoodbye();
+                toReturn = "Bye!";
+            } else if (Parser.commandEquals("list", input)) {
+                Ui.displayList(tasks);
+                toReturn = tasks.toString();
+            } else if (Parser.commandEquals("done", input)) {
+                int taskNo = Parser.getTaskNo(input);
+                tasks.markTaskAsDone(taskNo);
+
+                toReturn = "I have marked that task as done!";
+            } else if (Parser.commandEquals("delete", input)) {
+                int taskNo = Parser.getTaskNo(input);
+                tasks.deleteTask(taskNo);
+
+                toReturn = "I have deleted that task!";
+            } else if (Parser.commandEquals("deadline", input)) {
+                if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                if (tasks.isFull()) throw new TooManyTasksException();
+
+                tasks.addDeadline(Parser.getArgs(input));
+                toReturn = "I have added that deadline!";
+            } else if (Parser.commandEquals("event", input)) {
+                if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                if (tasks.isFull()) throw new TooManyTasksException();
+
+                tasks.addEvent(Parser.getArgs(input));
+
+                toReturn = "I have added that event!";
+            } else if (Parser.commandEquals("todo", input)) {
+                if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                if (tasks.isFull()) throw new TooManyTasksException();
+
+                tasks.addTodo(Parser.getArgs(input));
+                toReturn = "I have added that todo!";
+            } else if (Parser.commandEquals("find", input)) {
+                if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+
+                toReturn = tasks.findToString(Parser.getArgs(input));
+            } else {
+                throw new UnknownCommandException();
+            }
+
+            // Rewrites the entire file for every update you make here
+            // Probably O(n^2) time where n is the number of tasks but this is the simplest change we can make
+            storage.writeTasks(tasks);
+            return toReturn;
+        } catch (DukeException e) {
```

Comment:
nice that you had a exception class that is used to catch known exceptions that happen in your program. however, you could consider adding another catch block for the general Exception class. this is just in case your program runs into an exception that you did not foresee and causes the program to crash. inside this block you could pass a friendly message to the end user to inform them that something bad has happened.